Docs: clarify DEWP asset file naming rule (#80620)
WordPress/gutenberg: packages/dependency-extraction-webpack-plugin/README.md

Each entry point in the webpack bundle will include an asset file that declares the WordPress script dependencies that should be enqueued. This file also contains the unique version hash calculated based on the content of the entry point's files, including any extracted styles.

> The asset file name mirrors the emitted JavaScript file name, replacing the `.js` (or `.mjs`) extension with `.asset.php` (or `.asset.json` when `outputFormat` is `json`). Because it follows the webpack `output.filename` setting, a single unnamed entry produces `main.asset.php` by default, and an output such as `index.min.js` produces `index.min.asset.php`.

Reviewer: ```suggestion
The asset file name mirrors the emitted JavaScript file name, replacing the file extension with `.asset.php` or `.asset.json` depending on the configured `outputFormat`. Because it follows the webpack `output.filename` setting, a single unnamed entry produces `main.asset.php` by default, and an output such as `index.min.js` produces `index.min.asset.php`.
```

Author: Applied, thanks, that's clearer.

> The asset file name mirrors the emitted JavaScript file name, replacing the `.js` (or `.mjs`) extension with `.asset.php` (or `.asset.json` when `outputFormat` is `json`). Because it follows the webpack `output.filename` setting, a single unnamed entry produces `main.asset.php` by default, and an output such as `index.min.js` produces `index.min.asset.php`.

Author: Good catch, dropping the `.mjs` mention from that section since it doesn't apply when `output.module` is false.
